### ISSUE TITLE
CA: UCSD Results Staging EHR Filing

### ISSUE THREAD

**Issue by JohnNKing** [CDCgov/trusted-intermediary#1229]
# Story
As UCSD, so that I can review and provide patient care based on NBS findings, I need sample result messages sent by ReportStream staging to appear to clinicians in my EHR test environment.

## Pre-conditions
- Out of scope: Production connectivity

## Tasks
- [x] Review old ticket #1222
- [x] Complete initial round trip test with CDPH and UCSD (planned for 10/3/2024)
- [x] Confirm 'µ' character in a result observation was not corrupted
- [x] Get confirmation from UCSD that the sample result files and displays successfully (without any manual modification)

## Notes
- *Optional: Any reference material or thoughts we may need for later reference*

**Comment by basiliskus** [comment]
We could add the correct encoding of `µ` as part of the tasks or AC

**Comment by JohnNKing** [comment]
> We could add the correct encoding of `µ` as part of the tasks or AC

I know we also #1222 -- I could make resolving that bug a precondition for this story if it would help.

**Comment by jorg3lopez** [comment]
Update:
There will be a live testing session on Wednesday, Oct 2nd. In this live session, we plan to bring up the issue with the mu character. We will ask how the value shows up in UCSD's system. This will let us know if there are any issues or if everything is working as expected.

**Comment by jorg3lopez** [comment]
Update:
During the live test, we noticed that the mu character was not showing properly. We will need to do more research and see what encoding (ISO, UTF) UCSD Epic is using.

**Comment by basiliskus** [comment]
I'll do a round of testing with RS to make sure the message is sent with the mu character showing correctly

**Comment by basiliskus** [comment]
After some tests I found a couple of things. While testing locally, I found the files are correctly encoded throughout the flow. While testing in staging, I found that somewhere between RS receiving the initial HL7 message and sending the FHIR message to TI, the encoding gets garbled. So it's confirmed RS is sending the message to UCSD with the 'µ' character missing

**Comment by basiliskus** [comment]
Created a bug report in RS for the encoding issue here: https://github.com/CDCgov/prime-reportstream/issues/16196

**Comment by jbiskie** [comment]
We changed the locale to UTF-8 in the dockerfile configuration. No impact.

**Comment by jbiskie** [comment]
Removing `-Dfile.encoding=UTF-8` in the `JAVA_OPTS` in `local.settings.json` breaks local deployments.

**Comment by basiliskus** [comment]
It may be worth reopening https://github.com/CDCgov/trusted-intermediary/issues/1222 because it seems there's an actual bug in Daily Data regarding the encoding of the 'µ' character, that is separate from the encoding issue currently being fixed in the deployed environments

**Comment by basiliskus** [comment]
https://github.com/CDCgov/trusted-intermediary/issues/1222 reopened

**Comment by basiliskus** [comment]
[This open PR](https://github.com/CDCgov/reportstream-sftp-ingestion/pull/193) in the reportstream-sftp-ingestion
repository fixes the encoding issue when sending the ORU using the SFTP service. Once the PR is merged, we need to test uploading new files to be sent in staging and checking the EHR Filing with UCSD

**Comment by basiliskus** [comment]
The encoding fix in the reportstream-sftp-ingestion service has been merged and tested in staging. We should be ready to test this issue in the staging EHR Filing

**Comment by scleary1cs** [comment]
Closing in favor of #1442 .  Moving AC over.
